volgen/volgen.go

```diff
@@ -52,16 +58,42 @@ func GenerateClientVolfile(vol *volume.Volinfo) error {
 		return err
 	}
 
-	buf := new(bytes.Buffer)
-	if err := cg.Write(buf); err != nil {
-		return err
-	}
-	if _, err := store.Store.Put(context.TODO(), volfilePrefix+vol.Name, buf.String()); err != nil {
-		return err
-	}
+	return cg.WriteToFile(getGfproxydVolFilePath(vol.Name))
+}
+
+// GenerateClientVolfile generates the client volfile and stores it in etcd
+func GenerateClientVolfile(vol *volume.Volinfo) error {
```

Comment:
It will be better to refactor this into a couple of functions, `GenerateClientVolfiles` and `GenerateClientVolfile`.

`GenerateClientVolfile` should accept volinfo, template, volfile path and extra map as arguments. `GenerateClientVolfiles` can then call `GenerateClientVolfile` with required params.

This applies to `DeleteClientVolfile` as well.

Reply:
Done